sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/KalixTestKit.java

```diff
@@ -87,7 +93,22 @@ public static class Settings {
      */
     @Deprecated
     public Settings(final Duration stopTimeout) {
-      this(stopTimeout, "self", false, false, Optional.empty(), Collections.emptyMap());
+      this(stopTimeout, "self", false, false, Optional.empty(), Collections.emptyMap(), EventingSupport.GRPC_BACKEND);
+    }
+
+    public enum EventingSupport {
+      /**
+       * This is the default type used and allows the testing eventing integrations without an external broker dependency
+       * running.
+       */
+      GRPC_BACKEND,
+
+      /**
+       * Used if you want to use an external Google PubSub Emulator on your tests.
+       *
+       * Note: the Google PubSub Emulator need to be started independently.
+       */
+      GOOGLE_PUBSUB_EMULATOR
```

Comment:
What can I do if I'm using Kafka? 

Maybe we should use different names here. Something like: `TEST_BROKER` and `EXTERNAL_BROKER`.

Also, for the users, the fact is a GRPC_BACKEND is not relevant. That's an implementation detail on our side.

Reply:
Now, we can go one step further and decide that the testkit always uses a test broker and let users use the `DockerIntegrationTest` that Andrzej added recently for tests using external brokers. 

Anyway, just an idea for later. `DockerIntegrationTest` is not yet open for general use.

Reply:
The idea here was to allow clients that were using the googlepubsub in integration tests to still have the option to continue using it, even if now they need to pass in that config. Kafka was not supported for integration tests AFAIK.. or was it? My idea was if we wanted to support Kafka we would add a new option there and add the defaults and way to override the port by config, etc. But that would be a different issue.

Agree on `GRPC_BACKEND`, `TEST_BROKER` seems like a good one. 👍

Reply:
Correct, testing with external Kafka is not supported.
If we move further with `DockerIntegrationTest` idea, we may decide that Kafka tests should rely on docker-compose instead.